src/app/server/OnboardingCodesUtil.cpp

```diff
@@ -122,6 +123,25 @@ CHIP_ERROR GetQRCode(std::string & QRCode, chip::RendezvousInformationFlags rend
     return err;
 }
 
+CHIP_ERROR GetQRCodeUrl(char * QRCodeUrl, std::string & QRCode, size_t maxSize)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    int writtenDataSize;
+
+    VerifyOrExit(QRCodeUrl, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(maxSize >= (strlen(kQrCodeBaseUrl) + strlen(kUrlDataAssignmentPhrase) + 3 * QRCode.size() + 1),
```

Comment:
We shouldn't require a buffer for the worst-case. In case there's no special character a buffer equal to `strlen(kQrCodeBaseUrl) + strlen(kUrlDataAssignmentPhrase) + QRCode.size() + 1` should be enough.

Reply:
Yeah, basically I just copied the required size from there https://github.com/kkasperczyk-no/connectedhomeip/pull/29/files/78f72155ec467bb628fae1176232a8ffd689a6c6#diff-acd4fd1ff3ccaac7692e2e3c20a8f4813f742bc11ab483ce160942ad6b5787b0L44 but I was curious why is it defined that way. Nevertheless I agree that size proposed by you should be enough.